libs/pageserver_api/src/controller_api.rs

```diff
@@ -48,6 +48,46 @@ pub struct TenantPolicyRequest {
     pub scheduling: Option<ShardSchedulingPolicy>,
 }
 
+impl FromStr for PlacementPolicy {
+    type Err = anyhow::Error;
+
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        match s {
+            "detached" => Ok(Self::Detached),
+            "secondary" => Ok(Self::Secondary),
+            _ if s.starts_with("attached:") => {
+                let mut splitter = s.split(':');
+                let _prefix = splitter.next().unwrap();
+                match splitter.next().and_then(|s| s.parse::<usize>().ok()) {
+                    Some(n) => Ok(Self::Attached(n)),
+                    None => Err(anyhow::anyhow!(
+                        "Invalid format '{s}', a valid example is 'attached:1'"
+                    )),
+                }
+            }
+            _ => Err(anyhow::anyhow!(
+                "Unknown placement policy '{s}', try detached,secondary,attached:<n>"
+            )),
+        }
+    }
+}
+
+impl FromStr for ShardSchedulingPolicy {
+    type Err = anyhow::Error;
+
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        match s {
+            "active" => Ok(Self::Active),
+            "essential" => Ok(Self::Essential),
+            "pause" => Ok(Self::Pause),
+            "stop" => Ok(Self::Stop),
+            _ => Err(anyhow::anyhow!(
+                "Unknown scheduling policy '{s}', try active,essential,pause,stop"
+            )),
+        }
+    }
+}
```

Comment:
nit: philosophically, I think these should be impls on a newtype that sits in the storcon_cli crate as these these FromStr's are the CLI's representation.trait to implement

Reply:
Yeah, it's nicer to put them in the CLI code to avoid risk that they're mis-used elsewhere.  moved in 7b6a6023b9bdd23c15b000f45f97384e9e93bd3d